src/ngraph/runtime/gpu/gpu_cuda_kernel_emitters.cpp

```diff
@@ -24,27 +24,29 @@ using namespace ngraph;
 using namespace ngraph::runtime::gpu;
 
 void runtime::gpu::emit_broadcast(const std::string& name,
+                                  std::array<std::string, 2> data_types,
+                                  GPURuntimeContext* ctx,
                                   CUdeviceptr in,
                                   CUdeviceptr out,
-                                  const std::array<std::string, 2>& data_types,
                                   size_t repeat_size,
                                   size_t repeat_times,
                                   size_t count)
 {
     std::string name_signature = name + "_" + data_types[0] + "_" + data_types[1];
     std::replace(name_signature.begin(), name_signature.end(), ' ', '_');
     // Create an instance of nvrtcProgram with the code string.
-    if (CudaFunctionPool::instance().get(name_signature) == nullptr)
+    auto compiled_kernel = ctx->nvrtc_cache->get(name_signature);
```

Comment:
maybe call it `compiled_kernel_pool` instead of `nvrtc_cache`? easier to understand, not everyone know what nvrtc is.